File: Frontend/src/common/Domain/EstateType.ts
export enum EstateType {
    House = 1,
    Apartment = 2,
    Office = 3,
    Shop = 4,
    Warehouse = 5,
    Land = 6
}

export const showRoomField = (estateType: EstateType) => {
    switch(estateType){
        case EstateType.House:
        case EstateType.Apartment:
            return true
        default:
            return false
    }
}

File: Backend/Domain/Enum/EstateType.cs
using System.ComponentModel;

namespace Domain.Enum
{
    public enum EstateType
    {
        [Description("House")]
        House = 1,

        [Description("Apartment")]
        Apartment = 2,

        [Description("Office")]
        Office = 3,

        [Description("Shop")]
        Shop = 4,

        [Description("Werehouse")]
        Werehouse = 5,

        [Description("Land")]
        Land = 6
    }
}
